src/dcrypto/v1/v1_test.go

```diff
@@ -92,3 +94,37 @@ func TestRoundTrip(t *testing.T) {
 		}
 	}
 }
+
+func TestHash(t *testing.T) {
+	sizes := []int{24, 1024, 15872, 16364, 16384, 16394, 16896, 66560}
+	for _, size := range sizes {
+		h := md5.New()
```

Comment:
I forgot to ask if you think you'll need this logic? I was trying to figure out how to solve the issue of determining what has changed so did this. Now you could potentially check localsize+overhead and modTime and then if you were asked to do a checksum you could get the first maxheadersize as a partial download and hash the local file with that information. If the hash isn't something that is needed I could revert that change.